container_registry/container_analysis/samples.py

```diff
@@ -0,0 +1,343 @@
+            to a retryable error and retry attempts failed.
+        ValueError: If the parameters are invalid.
+    """
+    client = container_analysis_client.ContainerAnalysisClient()
+    note_name = client.note_path(project_id, note_id)
+
+    response = client.list_note_occurrences(note_name)
+    count = 0
+    for o in response:
+        # do something with the retrieved occurrence
+        # in this sample, we will simply count each one
+        count += 1
+    return count
+# [END occurrences_for_note]
+
+
+# [START occurrences_for_image]
+def get_occurrences_for_image(image_url, project_id):
+    """Retrieves all the occurrences associated with a specified image
+
+    :param note_id: the note's unique identifier
+    :param project_id: the project's unique identifier
+    :return: number of occurrences found
+    :raises
+        google.api_core.exceptions.GoogleAPICallError: If the request
+            failed for any reason.
+        google.api_core.exceptions.RetryError: If the request failed due
+            to a retryable error and retry attempts failed.
+        ValueError: If the parameters are invalid.
+    """
+    filterStr = "resourceUrl=\"" + image_url + "\""
+    client = container_analysis_client.ContainerAnalysisClient()
+    project_name = client.project_path(project_id)
+
+    response = client.list_occurrences(project_name, filter_=filterStr)
+    count = 0
+    for o in response:
+        # do something with the retrieved occurrence
+        # in this sample, we will simply count each one
+        count += 1
+    return count
+# [END occurrences_for_image]
+
+
+# [START pubsub]
+def pubsub(subscription_id, timeout, project_id):
+    """Handle incoming occurrences using a pubsub subscription
+
+    :param subscription_id: the user-specified identifier for the subscription
+    :param timeout: the number of seconds to listen for pubsub messages
+    :param project_id: the project's unique identifier
+    :return: number of occurrence pubsub messages received
+    """
+    client = SubscriberClient()
+    subscription_name = client.subscription_path(project_id, subscription_id)
+    receiver = MessageReceiver()
+    client.subscribe(subscription_name, receiver.pubsub_callback)
+
+    # listen for 'timeout' seconds
+    print("listening")
+    for _ in range(timeout):
+        sleep(1)
+    # print and return the number of pubsub messages received
+    print(receiver.msg_count)
+    return receiver.msg_count
+
+
+class MessageReceiver:
+    """Custom class to handle incoming pubsub messages
+    In this case, we will simply print and count each message as it comes in
+    """
+    def __init__(self):
+        # initialize counter to 0 on initialization
+        self.msg_count = 0
+
+    def pubsub_callback(self, message):
+        # every time a pubsub message comes in, print it and count it
+        self.msg_count += 1
+        print("Message " + str(self.msg_count) + ": " + message.data)
```

Comment:
Prefer format to string concatenation:

    print('Message {}: {}'.format(self.msg_count, message.data)